Description: Sync latest strutils from oslo-incubator for mask_password fix
 This sync pulls in:
   1131b56 Enable mask_password to handle byte code strings
 .
 This is the only update since the last sync:
   beca4db Sync latest strutils from oslo-incubator
 .
 This is needed to fix a delete volume failure when processutils
 calls mask_password with a byte code string.
Author: James Carey <jecarey@us.ibm.com>
Origin: upstream, https://review.openstack.org/#/c/129613/
Date: Tue, 16 Sep 2014 14:13:51 +0000 (+0000)
X-Git-Url: https://review.openstack.org/gitweb?p=openstack%2Fcinder.git;a=commitdiff_plain;h=3821206ff1455ffc682739911f7f59bdde6e18d2
Change-Id: Ie8a8c8e26abc0e387830bc9246e9a44913ae9b24
Bug-Ubuntu: https://launchpad.net/bugs/1368527

diff --git a/cinder/openstack/common/strutils.py b/cinder/openstack/common/strutils.py
index e3e0b76..381ca3e 100644
--- a/cinder/openstack/common/strutils.py
+++ b/cinder/openstack/common/strutils.py
@@ -297,7 +297,12 @@ def mask_password(message, secret="***"):
     >>> mask_password("u'original_password' :   u'aaaaa'")
     "u'original_password' :   u'***'"
     """
-    message = six.text_type(message)
+    try:
+        message = six.text_type(message)
+    except UnicodeDecodeError:
+        # NOTE(jecarey): Temporary fix to handle cases where message is a
+        # byte string.   A better solution will be provided in Kilo.
+        pass
 
     # NOTE(ldbragst): Check to see if anything in message contains any key
     # specified in _SANITIZE_KEYS, if not then just return the message since
